### DISCUSSION TITLE
Brazilian Market on WooCommerce doesn't work

### DISCUSSION THREAD

**Discussion by robertoduha** [woocommerce/woocommerce#51932] [Checkout Flow]
os campos CPF e CNPJ não aparecem no cadastro e não fazem falta na finalização da compra e pagamento!
...
The CPF and CNPJ fields do not appear in the registration and are not required when completing the purchase and payment!

**Comment by alexflorisca** [comment]
Hi there, it looks to me like you'd like some support with WooCommerce. Please reach out in [Slack](https://woocommerce.com/community-slack/) or the [WooCommerce Forums](https://wordpress.org/support/plugin/woocommerce/)

**Comment by qu4rta** [comment]
It should be implemented natively in WC.
Depending on an outdated third party is not good.

**Reply by alexflorisca** [reply]
I'm not familiar with the CPF and CNPJ fields - what are they?

**Reply by qu4rta** [reply]
They are like the american "Social Security Number". Every persons has an unique document number that since 2022 is required for shipping.
There's a plugin available to show the fields on the checkout page, but it is not updated to the new block version of the checkout page, and we have to stay with the old shortcode version for it to work.

**Comment by alexflorisca** [comment]
Gotcha. I think many countries are in your situation, where they need to show extra fields depending on regulation. I think this is something that would be quite useful to have in core, at least for the bigger market places like Brazil, some European countries. What do you think @ralucaStan?

**Reply by qu4rta** [reply]
I'll be happy to provide any insights you may need to clarify their functions. Just let me know how to reach you privately and I can give you examples of what I'm saying if you think it may help.

**Reply by ralucaStan** [reply]
Putting this on @pmcpinto's radar

**Reply by pmcpinto** [reply]
> I'll be happy to provide any insights you may need to clarify their functions. Just let me know how to reach you privately and I can give you examples of what I'm saying if you think it may help.

@qu4rta can you share more examples? To ensure we cover the most impactful use cases. Thank you!

**Reply by qu4rta** [reply]
It's mandatory since 2022 in Brazil that every shipping has to have the CPF (for persons) or CNPJ (for businesses) of the buyer and the seller.

Today in Woocommerce/Wordpress the users has to rely on a outdated plugin that make these fields available on the back-end. It doesn't work on the new checkout page with blocks, only on the older version with shortcodes.

If this field isn't showing on the checkout page, buyers won't be able to receive their packages because it will be empty and as I said, it's mandatory now to ship stuff.

In my point of view, it would make more sense to have it under the account fields on the checkout/register (Name, e-mail, phone, CPF/CNPJ) and not on the shipping, because sometimes people may be buying stuff as gift and won't have this info of the person who'll receive it.

![Captura de tela 2024-11-27 111041](https://github.com/user-attachments/assets/2af62b2e-b68a-439c-a260-5e929d8e2a7e)

This is how I use it on my store, with the older version of checkout using the plugin (I couldn't make it show on the new version of checkout).

It would be amazingly helpful to have this field in core (or the possibility to have a native option to add custom fields on checkout maybe, as other countries could use something similar).

I'm linking the plugin required for Brazil market that needs to be installed so the store works.
https://pt.wordpress.org/plugins/woocommerce-extra-checkout-fields-for-brazil/

**Comment by pierorocca** [comment]
> Gotcha. I think many countries are in your situation, where they need to show extra fields depending on regulation. I think this is something that would be quite useful to have in core, at least for the bigger market places like Brazil, some European countries. What do you think @ralucaStan?

100% support this. These are localization requirements that are not being met and Brazil has had these requirements for at least the last 10 years. Similar to EU Vat Tax and anything else that's mandatory and a regulatory requirement, it needs to be supported in core. There's zero doubt or ambiguity. It'd be great to remove the need for 3P plugins or having to resort to Checkout Field Editor or the Additional Fields API.

**Reply by pierorocca** [reply]
Is there an open enhancement issue related to tis ticket? I don't see anything linked here.

**Reply by ralucaStan** [reply]
There isn't a specific issue for this ticket so feel free to open a new one

**Comment by robertoduha** [comment]
ok
thankyou


_____________________________________
Roberto Duha
(11) 96725-9702 Whatsapp (Op. Oi)

**não verifico mensagens de voz nos celulares.*


Em qui., 16 de jan. de 2025 às 03:55, Piero Rocca ***@***.***>
escreveu:

**Comment by pierorocca** [comment]
https://github.com/woocommerce/woocommerce/issues/54580
